test/utils/test_sampling.py

```diff
@@ -616,15 +623,16 @@ def test_optimize_posterior_samples(self):
             self.assertTrue(torch.all(X_opt >= bounds[0]))
             self.assertTrue(torch.all(X_opt <= bounds[1]))
 
-            # Check that the all found optima are larger than the observations
-            # This is not 100% deterministic, but just about.
             Y_queries = paths(X)
             # this is when we negate, so the values should be smaller
             if posterior_transform:
                 self.assertTrue(torch.all(f_opt < Y_queries.min(dim=-2).values))
 
             # otherwise, larger
             else:
+                print(torch.all(f_opt > Y_queries.max(dim=-2).values))
+                print(f_opt - Y_queries.max(dim=-2).values)
+                breakpoint()
```

Comment:
This needs cleaning up

Reply:
Oh shoot, that shouldn't have made it in there. Thanks.